tests/model/test_modulemd_model.py

```diff
@@ -0,0 +1,77 @@
+import sys
+import os
+
+import pytest
+
+from pubtools.pulplib import (
+    ModulemdUnit,
+    ModulemdDefaultsUnit,
+)
+from frozendict import frozendict
+from frozenlist2 import frozenlist
+
+
+def get_test_values():
+
+    default_unit = ModulemdDefaultsUnit(
+        unit_id="md_unit_id",
+        name="md_unit_name",
+        repo_id="md_unit_repo",
+        profiles={"1.0": ["default"]},
+    )
+
+    md_unit = ModulemdUnit(
+        unit_id="unit_id",
+        name="name",
+        stream="1.0",
+        version=1,
+        context="context",
+        arch="x86_64",
+        profiles={
+            "default": {
+                "description": "Default description",
+                "rpms": ["rpm"],
+            },
+        },
+    )
+    return default_unit, md_unit
+
+
+def test_convert_profile_value():
+    """Profile and the values it contains should be immutable"""
+
+    default_unit, md_unit = get_test_values()
+
+    assert isinstance(default_unit.profiles, frozendict)
+    assert isinstance(default_unit.profiles["1.0"], frozenlist)
+    assert isinstance(md_unit.profiles, frozendict)
+    assert isinstance(md_unit.profiles["default"], frozendict)
+    assert isinstance(md_unit.profiles["default"]["rpms"], frozenlist)
+
+
+def test_profiles_have_hash():
+    """If two objects are equal, then they must have the same hashcode"""
+    default_unit_1, md_unit_1 = get_test_values()
+    default_unit_2, md_unit_2 = get_test_values()
+
+    assert hash(default_unit_1) == hash(default_unit_2)
```

Comment:
There are issues with this test on py2.  Hashability is one of the reasons we want to be immutable, but since we fall back to plain old dict on py2, the tests there accurately complain that it can't be hashed.

I guess there would be ways to fix this, but my suggestion for the time being would be to mark this test as xfail on python 2 only.  Hopefully, we manage to retire py2 support later in the year or next year and we never have to bother doing a more proper fix.